packages/protocol/contracts/identity/IdentityProxyHub.sol

```diff
@@ -0,0 +1,21 @@
+pragma solidity ^0.5.13;
+
+import "./IdentityProxy.sol";
+
+contract IdentityProxyHub {
+  mapping(bytes32 => IdentityProxy) public identityProxies;
+
+  function getIdentityProxy(bytes32 identifier) public returns (IdentityProxy) {
+    if (address(identityProxies[identifier]) == address(0)) {
+      identityProxies[identifier] = new IdentityProxy();
+    }
+
+    return identityProxies[identifier];
+  }
+
+  function makeCall(bytes32 identifier, address destination, uint256 value, bytes calldata data)
```

Comment:
Instead of having the ownership check being here, and needing an external call to the proxy contract, why not have the Identity proxy execute the logic and have the caller call it directly to save gas and a bit of contract complexity. (The caller will already need to be aware of the proxies address to use it for it's application, and it can look up that address as a view call if it doesn't have it)

Reply:
I was actually thinking about this as well. I don't feel particularly strongly and would be open to moving the heuristic to IdentityProxy instead.

The benefits of the current approach is less deployed bytecode in storage (at the cost of slightly higher gas spent over time) and that you only need to care about the single IdentityProxyHub address if all you're doing is making calls (you don't even need to know your proxy's address since currently `makeCall` will provision it/look it up for you if you don't know it), which could make some parts of applications simpler.

That said, I do agree that long term this will burn more gas, assuming we expect users to make many `makeCall`s. Guess that's not entirely true in the use case we're currently building for (account recovery), but could become true depending on what these contracts are going to be used for in the future.